srv/salt/ceph/stage/prep/minion/default.sls

```diff
@@ -19,23 +24,69 @@ common packages:
     - tgt: '*'
     - sls: ceph.packages.common
 
-updates:
+{% if salt['saltutil.runner']('cephprocesses.check', roles=['mon']) == True %}
+
+{% for host in salt.saltutil.runner('orderednodes.unique', cluster='ceph') %}
+
+wait until the cluster has recovered before processing {{ host }}:
+  salt.state:
+    - tgt: {{ salt['pillar.get']('master_minion') }}
+    - sls: ceph.wait
+    - failhard: True
+
+check if all processes are still running after processing {{ host }}:
   salt.state:
     - tgt: '*'
+    - sls: ceph.cephprocesses
+    - failhard: True
+
+# After the last item in the iteration was processed the reactor 
+# still sets ceph osd set noout. So setting this after is still necessary.
+unset noout after processing {{ host }}:
+  salt.state
+    - sls: ceph.noout.unset
+    - tgt: {{ salt['pillar.get']('master_minion') }}
+    - failhard: True
+
+updating {{ host }}:
+  salt.state:
+    - tgt: {{ host }}
+    - tgt_type: compound
     - sls: ceph.updates
+    - failhard: True
 
-restart:
+check if restart is needed for {{ host }}:
```

Comment:
> I think I misinterpreted the code yesterday. The PackageManager class already does a reboot if the package manager requests it, yeah?

Yes, with one caveat. currently `zypper up` is used, while only `zypper patch` yields the correct returncode in order to determine the need for a reboot. Will be changed with two (default, non-default sls files). 

> And this reboot is not related to package installation?

It is. Based on the returncode(for zypper) and on the /var/something/reboot-required on (deb based) the module will trigger a reboot. But the packages(or specfiles) themselves tell the `Packagemanager` if a reboot is required or not (not sure on that one).